nvm.sh

```diff
@@ -3227,11 +3239,13 @@ nvm_auto() {
       nvm install >/dev/null
     fi
   elif [ "_$NVM_MODE" = '_use' ]; then
-   VERSION="$(nvm_resolve_local_alias default 2>/dev/null || nvm_echo)"
-    if [ -n "$VERSION" ]; then
-      nvm use --silent "$VERSION" >/dev/null
-    elif nvm_rc_version >/dev/null 2>&1; then
-      nvm use --silent >/dev/null
+    if [ "_$NVM_CURRENT" = '_system' ]; then
```

Comment:
I'll move it into a separate PR, however I think doing it in `nvm_auto` is better than `nvm_process_parameters`, because:

1. Checking whether nvm has already run is not *processing a parameter*.
1. Currently there's only 1 version check being done. Moving this to `nvm_process_parameters` means that either we'll should do a separate version check there (performance impact), or we *move* the version check to `nvm_process_parameters` and feed the result to `nvm_auto` as a parameter, to keep the performance impact down, but this kinda makes `nvm_auto` feel less "automatic" if it doesn't search the version by itself.
1. Checking the version and deciding whether or not to load up a different version of node seems appropriate for the "automatic" functionality.